lib/types/thermostat.js

```diff
@@ -0,0 +1,345 @@
+                params[action_param] = value;
+                this.executeVeraAction(params); //Set Heat setpoint
+                params["serviceId"] = serviceIdCool;
+                return this.executeVeraAction(params); //Set Cool setpoint
+            },
+
+            getTemperatureDisplayUnits: function(){
+                return temperatureDisplayUnit;
+            },
+
+            getBatteryLevel: function () {
+                if (this.states && this.states['BatteryLevel']) {
+                    return parseInt(this.states['BatteryLevel']);
+                }
+                return 50;
+            },
+
+            getIsLowBattery: function () {
+                var level = this.getBatteryLevel();
+                if (level && level < 20) {
+                    return Characteristic.StatusLowBattery.BATTERY_LEVEL_LOW;
+                }
+                return Characteristic.StatusLowBattery.BATTERY_LEVEL_NORMAL;
+            },
+
+            identify: function () {
+                debug("Identity of the thermostat is %s", device.name);
+            }
+        };
+
+        var thermostatUUID = uuid.generate('device:thermostat:' + config.cardinality + ':' + device.id);
+        var thermostat = new Accessory(device.name, thermostatUUID);
+        thermostat.username = functions.genMac('device:' + config.cardinality + ':' + device.id);
+        thermostat.deviceid = device.id;
+
+        thermostat.on('identify', function (paired, callback) {
+            Thermostat.identify();
+            callback(); // success
+        });
+
+        var service = new Service.Thermostat(device.name);
+
+        service.getCharacteristic(Characteristic.CurrentHeatingCoolingState)
+            .on('get', function (callback) {
+                debug('Getting current heating/cooling state for %s', device.name);
+                Thermostat.getCurrentHeatingCoolingState().then(function(val){
+                    // return our current value
+                    callback(null, val);
+                });
+            });
+
+        service.getCharacteristic(Characteristic.TargetHeatingCoolingState)
+            .on('get', function (callback) {
+                debug('Getting target heating/cooling value for %s', device.name);
+                Thermostat.getTargetHeatingCoolingState().then(function(val){
+                    // return our current value
+                    callback(null, val);
+                });
+            });
+
+        service.getCharacteristic(Characteristic.TargetHeatingCoolingState)
+            .on('set', function (value, callback) {
+                debug('Setting target heating/cooling value for %s', device.name);
+                Thermostat.setTargetHeatingCoolingState(value).then(function(val){
+                    // return our current value
+                    callback(null, val);
+                });
+            });
+
+        service.getCharacteristic(Characteristic.CurrentTemperature)
```

Comment:
Yeah. You'll only see the results when both the Vera and your phone are set to Fahrenheit. It effectively does the math again. Check out what homebride-nest did as an example.